src/managers/config.py

```diff
@@ -547,6 +560,41 @@ def metrics_reporter_properties(self) -> list[str]:
             username=ADMIN_USER, prefix="cruise.control.metrics.reporter"
         )
 
+    @property
+    def authorizer_class(self) -> list[str]:
+        """Return the authorizer Java class used on Kafka."""
+        if self.state.runs_controller:
+            return ["authorizer.class.name=org.apache.kafka.metadata.authorizer.StandardAuthorizer"]
+        return ["authorizer.class.name=kafka.security.authorizer.AclAuthorizer"]
+
+    @property
+    def controller_properties(self) -> list[str]:
+        """Builds all properties necessary for starting Kafka controller service.
+
+        Returns:
+            List of properties to be set
+        """
+        # TODO: this check will need to be done on relation data
+        roles = "process.roles="
+        if self.state.runs_controller:
+            roles += self.config.roles
```

Comment:
**todo**: we need to remove `balancer` if it's present

```suggestion
        if self.state.runs_controller:
            roles += ",".join(set(self.config.roles.split(",")) - {"balancer"})
```

Reply:
Will address in an upcoming PR